### DISCUSSION TITLE
VScode using SDCC

### DISCUSSION THREAD

**Discussion by pensharpie** [microsoft/vscode-discussions#1295] [Extension Development QnA] [answered]
The following refers to the 8051, VSCode and SDCC

Linker options:  hex output --model-large --code-loc 0x0000 --code-size 0x6FF8 --xram-loc-xram-size 0x700

Q1.  Why isnt the code size and xram size set at 0xFFFF.  Why would the code and xram size be less than 0xFFFF?

Q2.  Does the EIDE debug extension file which generates a .cdb file add debug symbols to the executable file?  In gcc, debug symbols are normally stripped from the executable since the debug symbols greatly increases file size.

**Comment by isidorn** [comment, marked as answer]
This is a community for vscode extensino developers. For general vscode questions please ask them on stack overflow
